@xen-orchestra/web/src/components/vm/VmVifsTable.vue

```diff
@@ -0,0 +1,282 @@
+              <div v-else v-tooltip="{ placement: 'bottom-end' }" class="text-ellipsis">
+                {{ column.value }}
+              </div>
+            </td>
+          </tr>
+        </template>
+      </VtsDataTable>
+      <UiTopBottomTable :selected-items="0" :total-items="0" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useNetworkStore } from '@/stores/xo-rest-api/network.store'
+import { useVifStore } from '@/stores/xo-rest-api/vif.store.ts'
+import { useVmStore } from '@/stores/xo-rest-api/vm.store.ts'
+import type { XoVif } from '@/types/xo/vif.type.ts'
+import VtsConnectionStatus from '@core/components/connection-status/VtsConnectionStatus.vue'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTableActions from '@core/components/ui/table-actions/UiTableActions.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import useMultiSelect from '@core/composables/table/multi-select.composable'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import {
+  faAlignLeft,
+  faAt,
+  faCaretDown,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPlus,
+  faPowerOff,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { vifs } = defineProps<{
+  vifs: XoVif[]
+}>()
+
+const { get: getNetwork } = useNetworkStore().subscribe()
+const { get: getVm } = useVmStore().subscribe()
+const { isReady, hasError } = useVifStore().subscribe()
+const { t } = useI18n()
+
+const selectedVifId = useRouteQuery('id')
+
+const getNetworkName = (vif: XoVif) => {
+  const network = getNetwork(vif.$network)!
+
+  return network.name_label ? network.name_label : ''
+}
+
+const searchQuery = ref('')
+
+const filteredVifs = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return vifs
+  }
+
+  return vifs.filter(vif =>
+    [...Object.values(vif), getNetworkName(vif)].some(value => String(value).toLocaleLowerCase().includes(searchTerm))
+  )
+})
+
+const getIpAddresses = (vif: XoVif) => {
+  const vm = getVm(vif.$VM)
+
+  if (!vm) return []
+
+  if (!vm.addresses) return []
+
+  return [...new Set(Object.values(vm.addresses).sort())]
+}
```

Comment:
Maybe more concise like :

```suggestion
const getIpAddresses = (vif: XoVif) => {
  const addresses = getVm(vif.$VM)?.addresses
  
  return addresses ? [...new Set(Object.values(addresses).sort())] : []
}
```